### DISCUSSION TITLE
Word-level timings as a feature of the decoder?

### DISCUSSION THREAD

**Discussion by agne-ker** [espnet/espnet#4539] [Q&A] [answered]
Is there any plan to implement getting word-level timings from the decoder in ESPnet? I've seen in discussions that CTC segmentation can be used for getting word-level timings, but I am interested in getting them as a feature of the decoder.

**Comment by sw005320** [comment, marked as answer]
We do not have such a plan since (as you mentioned) two-pass CTC segmentation is enough for many applications. However, we are thinking of enhancing the time-synchronous decoding function and it may support the alignment function (but not word level, it would be BPE-token level).

**Reply by agne-ker** [reply]
Thanks for your answer. Do you have any timeline for the enhancement?

**Reply by sw005320** [reply]
We're planning to make a PR in August. 
Also, we can request it for RNN-T.
@b-flo, what do you think of adding a function of producing the BPE-token alignment? (not word).

**Reply by b-flo** [reply]
Well, token and word timing information is also something requested by my colleagues so yes definitely! In fact, I already did some work for that with the new version but it was put on standby for other things.

Btw, there are some interesting things in k2 to consider for timing information and forced alignment with transducer!
